src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -130,18 +147,39 @@ public void onAuthConfirmCall(@RequestBody LinkOnConfirmResponse linkOnConfirmRe
    *     linking.
    */
   @PostMapping({"/v0.5/links/link/on-add-contexts"})
-  public void onAddCareContext(
-      @RequestBody LinkOnAddCareContextsResponse linkOnAddCareContextsResponse) {
+  public ErrorResponse onAddCareContext(
+      @RequestBody LinkOnAddCareContextsResponse linkOnAddCareContextsResponse)
+      throws IllegalDataStateException {
     log.debug(linkOnAddCareContextsResponse.toString());
     if (linkOnAddCareContextsResponse != null) {
       if (linkOnAddCareContextsResponse.getError() != null) {
-        log.error(
-            "/v0.5/links/link/on-add-contexts error: "
-                + linkOnAddCareContextsResponse.getError().getMessage());
+        updateRequestError(
+            linkOnAddCareContextsResponse.getResp().getRequestId(),
+            "onAddCareContext",
+            linkOnAddCareContextsResponse.getError().getMessage());
+      } else {
+        requestLogService.setHipOnAddCareContextResponse(linkOnAddCareContextsResponse);
       }
-      requestLogService.setHipOnAddCareContextResponse(linkOnAddCareContextsResponse);
     } else {
-      log.error("Failed to add Context");
+      String error = "Got Error in onAddCareContext callback: gateway response was null";
+      updateRequestError(
+          linkOnAddCareContextsResponse.getResp().getRequestId(), "onAddCareContext", error);
```

Comment:
This is invoked when the `linkOnAddCareContextsResponse` is null and trying to fetch `linkOnAddCareContextsResponse.getResp().getRequestId()` might throw `NullPointerException`

Reply:
agree, made changes